[quant][graphmode] Add insert_prepack_unpack for conv2d

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27239 [refactor] Factor out insertPrepackUnpackForLinear
* #27238 [refactor] InsertObservers
* #27237 [quant][graphmode][fix] Make inserted child module names unique
* #27119 [quant][graphmode] Prepack folding for conv2d
* **#27118 [quant][graphmode] Add insert_prepack_unpack for conv2d**

Summary:
att

Test Plan:
test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D17717637](https://our.internmc.facebook.com/intern/diff/D17717637)

jerryzh168 committed:
[quant][graphmode] Add insert_prepack_unpack for conv2d

Summary:
att

Test Plan:
test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Add insert_prepack_unpack for conv2d"


Summary:
att

Test Plan:
test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -898,6 +921,8 @@ graph(%linear, %a_dequant, %w, %b, %w_scale, %w_zero_point, %w_dtype):
   rewriter.RegisterRewritePattern(linear_with_quant, linear_with_quant_prepack);
   rewriter.runOnGraph(graph, filter);
+
+  insertPrepackUnpackForConv2d(graph);
```
Code above should be factored out into a separate function (or `insertPrepackUnpackForConv2d` should be inlined here). Otherwise, it's inconsistent on the level of abstraction/details.

jerryzh168 replied:
yeah, planning to refactor the above into a different function as well, will do in a separate PR

ZolotukhinM reviewed (approved):
Looks good!

jerryzh168 committed:
Update on "[quant][graphmode] Add insert_prepack_unpack for conv2d"


Summary:
att

Test Plan:
test_jit.py

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
